ts/modules/marker-clusters.ts

```diff
@@ -0,0 +1,1948 @@
+            (this: Point, event: PointClickEventObject): void;
+        }
+        interface ClusterEventsDictionary {
+            onDrillToCluster: OnDrillToClusterCallbackFunction;
+        }
+        interface MarkerClusterOptions {
+            enabled?: boolean;
+            allowOverlap?: boolean;
+            minimumClusterSize?: number;
+            drillToCluster?: boolean;
+            layoutAlgorithm: LayoutAlgorithmOptions;
+            style?: StyleOptions;
+            dataLabels?: DataLabelsOptionsObject;
+            zones?: Array<ClusterZonesOptions>;
+            states?: PointStatesOptionsObject;
+            events?: ClusterEventsDictionary;
+        }
+        interface PointMarkerOptionsObject {
+            cluster?: MarkerClusterOptions;
+        }
+        interface PointOptionsObject {
+            lat?: number;
+            lon?: number;
+        }
+        interface SplittedDataObject {
+            index: number;
+            x: number;
+            y: number;
+            options?: (string | number | PointOptionsObject |
+            (string | number)[] | null);
+        }
+        interface SplittedDataArray extends Array<SplittedDataObject> {
+            posX?: number;
+            posY?: number;
+        }
+        interface SplittedData {
+            [key: string]: SplittedDataArray;
+        }
+        interface PointPositionObject {
+            x: number;
+            y: number;
+        }
+        interface PointClusterDistanceObject {
+            clusterIndex: number;
+            distance: number;
+        }
+        interface KmeansClusterObject {
+            posX: number;
+            posY: number;
+            oldX: number;
+            oldY: number;
+            startPointsLen: number;
+            points: Array<SplittedDataObject>;
+        }
+        interface ClusterAlgorithmsDictionary
+            extends Dictionary<Function>
+        {
+            grid(
+                processedXData: Array<number>,
+                processedYData: Array<number>,
+                visibleDataIndexes: Array<number>,
+                options: LayoutAlgorithmOptions
+            ): SplittedData;
+            kmeans(
+                processedXData: Array<number>,
+                processedYData: Array<number>,
+                visibleDataIndexes: Array<number>,
+                options: LayoutAlgorithmOptions
+            ): SplittedData;
+            optimizedKmeans(
+                processedXData: Array<number>,
+                processedYData: Array<number>,
+                visibleDataIndexes: Array<number>,
+                options: LayoutAlgorithmOptions
+            ): SplittedData;
+        }
+        interface PreventClusterCollisionsProps {
+            x: number;
+            y: number;
+            key: string;
+            groupedData: SplittedData;
+            gridSize: number;
+            defaultRadius: number;
+            clusterRadius: number;
+        }
```

Comment:
Name should be `MarkerClusterPreventCollisionObject`, but as this is only used for one function parameter, I doubt that the interface is needed at all. Why not use all properties directly as function parameters? What do you think, @TorsteinHonsi ?